tests/test_async.py

```diff
@@ -24,6 +24,20 @@ async def test_wait_during_not_done(sleeper):
         sleeper.sleep_for_a_bit())
 
 
+@pytest.mark.asyncio
+async def test_wait_until_done_with_delay(sleeper):
+    await asyncio.gather(
+        wait_until_awake_with_delay(sleeper),
+        sleeper.sleep_for_a_bit())
+
+
+@pytest.mark.asyncio
+async def test_wait_until_done_with_too_much_delay(sleeper):
```

Comment:
test name should describe the scenario, maybe a better name would be timeout_failure_of_async_condition or test_fail_on_timeout, etc.